tests/testthat/test-eval-time-single-selection.R

```diff
@@ -0,0 +1,242 @@
+# "selecting single eval time" means how functions like `show_best()` will pick
+# an evaluation time for a dynamic metric when none is given. Previously we
+# would find what is in the data and select a time that was close to the median
+# time. This was fine but inconsistent with other parts of tidymodels that do
+# similar operations. For example, tune_bayes has to have a metric to optimize
+# on so it uses the first metric in the metric set and, if needed, the first
+# evaluation time given to the function.
```

Comment:
```suggestion
```
I think this amount of context is better placed in the PR description (where you've already put it).